test/functional/outputs/otlp/forward_to_otlp_test.go

```diff
@@ -0,0 +1,192 @@
+package otlp
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	obs "github.com/openshift/cluster-logging-operator/api/observability/v1"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/test/framework/functional"
+	"github.com/openshift/cluster-logging-operator/test/helpers/otlp"
+	obstestruntime "github.com/openshift/cluster-logging-operator/test/runtime/observability"
+	"time"
+)
+
+const (
+	timestamp     = "2023-08-28T12:59:28.573159188+00:00"
+	timestampNano = "1693227568573159188"
+)
+
+var _ = Describe("[Functional][Outputs][Otlp] Functional tests", func() {
+	var (
+		framework    *functional.CollectorFunctionalFramework
+		appNamespace string
+	)
+
+	BeforeEach(func() {
+		framework = functional.NewCollectorFunctionalFramework()
+
+	})
+
+	AfterEach(func() {
+		framework.Cleanup()
+	})
+	Context("When forwarding logs to an otel-collector", func() {
+		It("should send application logs with correct grouping and resource attributes", func() {
+			obstestruntime.NewClusterLogForwarderBuilder(framework.Forwarder).
+				FromInput(obs.InputTypeApplication).
+				ToOtlpOutput()
+
+			Expect(framework.DeployWithVisitor(func(b *runtime.PodBuilder) error {
+				return framework.AddOTELCollector(b, string(obs.OutputTypeOTLP))
+			})).To(BeNil())
+
+			appNamespace = framework.Pod.Namespace
+
+			// Write message to namespace
+			crioLine := functional.NewCRIOLogMessage(timestamp, "Format me to a container message!", false)
+			Expect(framework.WriteMessagesToNamespace(crioLine, appNamespace, 1)).To(Succeed())
+			crioLine = functional.NewCRIOLogMessage(timestamp, "My second message.", false)
+			Expect(framework.WriteMessagesToNamespace(crioLine, appNamespace, 1)).To(Succeed())
+			crioLine = functional.NewCRIOLogMessage(timestamp, "My third and final message.", false)
+			Expect(framework.WriteMessagesToNamespace(crioLine, appNamespace, 1)).To(Succeed())
+
+			// Read logs
+			raw, err := framework.ReadRawApplicationLogsFrom(string(obs.OutputTypeOTLP))
+			Expect(err).To(BeNil(), "Expected no errors reading the logs for type")
+			Expect(raw).ToNot(BeEmpty())
+
+			logs, err := otlp.ParseLogs(raw[0])
+			Expect(err).To(BeNil(), "Expected no errors parsing the logs")
+
+			// Access the first (and only) ResourceLog
+			// Inspect resource attributes
+			resourceLog := logs.ResourceLogs[0]
+			resource := resourceLog.Resource
+			namespaceName := resource.FindStringValue(resource.NamespaceNameAttribute())
```

Comment:
Not to be solved here but we can add our "templates" here to automagically test these

Reply:
Certainly thanks!   Will be adding more tests over the next few days and can begin that process after.